caikit/runtime/model_management/model_manager.py

```diff
@@ -490,6 +507,35 @@ def _local_models_dir_sync(self, wait: bool = False):
                         exc_info=True,
                     )
 
+    def _model_write_in_progress(self, model_dir: str) -> bool:
+        """Returns true if model_dir is currently being written to. Uses the
+        runtime.lazy_load_write_detection_period_seconds configuration to sleep between
+        consecutive size checks of the directory.
+
+        Always returns false if runtime.lazy_load_write_detection_period_seconds is zero,
+        negative, or None.
+        """
+        if (
+            self._lazy_load_write_detection_period_seconds is None
+            or self._lazy_load_write_detection_period_seconds <= 0
+        ):
+            return False
+
+        # Get the current directory size
```

Comment:
How expensive is this? Is there any worry that it would be extremely expensive when "disk" is implemented as a shim to S3?

Reply:
Yeah idk!

I'm making the blind assumption that mounted filesystems like `s3fs` would do something smarter than a full remote read just to stat a file 🤷 

Might research it up a bit later